sdk/core/azure-core/CMakeLists.txt

```diff
@@ -120,6 +120,9 @@ target_include_directories(azure-core PUBLIC ${CURL_INCLUDE_DIRS})
 target_include_directories(azure-core INTERFACE ${nlohmann_json_INCLUDE_DIRS})
 target_link_libraries(azure-core INTERFACE CURL::libcurl Threads::Threads nlohmann_json::nlohmann_json)
```

Comment:
@antkmsft why was the `target_link_libraries` changed from private to interface as part of the vcpkg change?
https://github.com/Azure/azure-sdk-for-cpp/pull/1013/files#diff-c73316ca9adadb2b459c6af9255dfff8f388c81c73645ea73dcc8df61684dea8R118